enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -583,14 +584,23 @@ func (l *FileCacheLoader) cacheCOW(ctx context.Context, name string, remoteInsta
 				}
 			}
 
+			// If the chunk was mapped but the lazy digest is not set, it could not
+			// have been fetched remotely, because we use the lazy digest to do so.
+			// This means we must have generated the chunk from a local file
+			// Note: This has to be called before we call c.Digest(), which sets
+			// the lazy digest
+			chunkFetchedRemotely := c.Mapped() && c.LazyDigest() != nil
+
 			// Get or compute the digest.
 			d, err := c.Digest()
 			if err != nil {
 				return status.WrapError(err, "compute digest")
 			}
 			fn.Digest = d
 
-			if c.Mapped() {
+			// Don't cache artifacts that should already be in the cache
```

Comment:
I had started putting together a similar change earlier, but then remembered that we might want to cache the chunks just so that we can update the access time on the remote cache object.

Maybe there's some way we can explicitly update the access time if we think it should exist, and if it doesn't exist then fall back to re-uploading to remote cache?

Reply:
Oh... I guess this should be OK because when we fetch the remote snapshot tree, that updates the access time?

Reply:
We might want to update the atime in the local cache. Though I don't know if this is worth the complexity, if we can always fall back to fetching from remote cache